universe/Condition.cpp

```diff
@@ -9956,6 +9965,252 @@ unsigned int Not::GetCheckSum() const {
     return retval;
 }
 
+///////////////////////////////////////////////////////////
+// OrderedAlternativesOf
+///////////////////////////////////////////////////////////
+OrderedAlternativesOf::OrderedAlternativesOf(
+    std::vector<std::unique_ptr<ConditionBase>>&& operands) :
+    ConditionBase(),
+    m_operands(std::move(operands))
+{}
+
+bool OrderedAlternativesOf::operator==(const ConditionBase& rhs) const {
+    if (this == &rhs)
+        return true;
+    if (typeid(*this) != typeid(rhs))
+        return false;
+
+    const OrderedAlternativesOf& rhs_ = static_cast<const OrderedAlternativesOf&>(rhs);
+
+    if (m_operands.size() != rhs_.m_operands.size())
+        return false;
+    for (unsigned int i = 0; i < m_operands.size(); ++i) {
+        CHECK_COND_VREF_MEMBER(m_operands.at(i))
+    }
+
+    return true;
+}
+
+void OrderedAlternativesOf::Eval(const ScriptingContext& parent_context,
+                                 ObjectSet& matches, ObjectSet& non_matches,
+                                 SearchDomain search_domain/* = NON_MATCHES*/) const
+{
+    std::shared_ptr<const UniverseObject> no_object;
+    ScriptingContext local_context(parent_context, no_object);
+
+    if (m_operands.empty()) {
+        ErrorLogger() << "OrderedAlternativesOf::Eval given no operands!";
+        return;
+    }
+    for (auto& operand : m_operands) {
+        if (!operand) {
+            ErrorLogger() << "OrderedAlternativesOf::Eval given null operand!";
+            return;
+        }
+    }
+
```

Comment:
Before explaining the Not case, should probably explain the normal case...

For OrderedAlternativesOf [ A B C ], if any candidate matches A, then all candidates that match A are matched, or if no candidate matches A but any candidate matches B, then all candidates that match B are matched, or if no candidate matches A or B but any candidate matches C, then all candidates that match C are matched. If no candidate matches A, B, or C, then nothing is matched.

Then for the Not case...

For Not OrderedAlternativesOf [ A B C ], if any candidate matches A, then all candidates that do not match A are matched, or if no candidates match A but any candidate matches B, then all candidates that do not match B are matched, or if no candidate matches A or B but any candidate matches C, then all candidates that do not match C are matched. If no candidate matches A, B, or C, then all candidates are matched.